esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -399,52 +399,25 @@ int CFileSpraySoapBindingEx::downloadFile(IEspContext &context, CHttpRequest* re
         request->getParameter("Path", pathStr);
         request->getParameter("Name", nameStr);
         request->getParameter("DropZoneName", dropZoneName);
+        if (nameStr.isEmpty())
+            throw makeStringException(ECLWATCH_INVALID_INPUT,"File name not specified.");
         if (containsRelPaths(nameStr))
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid name %s", nameStr.str());
-
-        SecAccessFlags permission = getDZPathScopePermissions(context, dropZoneName, pathStr, netAddressStr);
-        if (permission < SecAccess_Read)
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s %s not allowed for user %s (permission:%s). Read Access Required.",
-                dropZoneName.str(), netAddressStr.str(), pathStr.str(), context.queryUserId(), getSecAccessFlagName(permission));
-#if 0
-        StringArray files;
-        IProperties* params = request->queryParameters();
-        Owned<IPropertyIterator> iter = params->getIterator();
-        if (iter && iter->first())
-        {
-            while (iter->isValid())
-            {
-                const char *keyname=iter->getPropKey();
-                if (!keyname || strncmp(keyname, "Names", 5))
-                    continue;
-
-                files.append(params->queryProp(iter->getPropKey()));
-                iter->next();
-            }
-        }
-#endif
-
-        if (netAddressStr.length() < 1)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Network address not specified.");
-
-        if (pathStr.length() < 1)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Path not specified.");
-
-        if (nameStr.length() < 1)
-            throw MakeStringException(ECLWATCH_INVALID_INPUT,"File name not specified.");
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file name %s", nameStr.str());
 
+        if (pathStr.isEmpty())
+            throw makeStringException(ECLWATCH_INVALID_INPUT, "Path not specified.");
         char pathSep = '/';
-        if ((osStr.length() > 1) && (atoi(osStr.str())== OS_WINDOWS))
-        {
+        if (osStr.isEmpty() && (atoi(osStr.str())== OS_WINDOWS))
```

Comment:
I think you mean "!osStr.isEmpty()"?  For the atoi the string shouldn't be empty.